### PR TITLE
Node version module update (PR on different branch)

### PR DESCRIPTION
**Pull request by KayLotura2**
# Issue
Bug with using file scraping to find `newrelic` Node Agent Module Version.
Updated to use already implemented payload.

# Goals
Cuts down extra work, returns more reliably accurate info.

# Implementation Details
Small changes, unit tests.

# How to Test
N/A, unit tests cover the small change.

### CODE REVIEW

File: tasks/node/agent/version.go [hunk 1]
```diff
@@ -36,63 +38,46 @@ func (t NodeAgentVersion) Dependencies() []string {
...
-		return result
-	}
-	logs, ok := upstream["Base/Log/Copy"].Payload.([]logtask.LogElement)
-	if !ok {
-		result.Status = tasks.None
-		result.Summary = "Node agent version not detected"
-		return result
+		return tasks.Result{
+			Status:  tasks.None,
+			Summary: "Node agent not detected",
```
Comment (uturunku1) [inline tasks/node/agent/version.go:47]: Here it should read: "Node Agent not detected. This task did not run"

File: tasks/node/agent/version.go [hunk 2]
```diff
@@ -36,63 +38,46 @@ func (t NodeAgentVersion) Dependencies() []string {
...
-			result.Status = tasks.Info
-			result.Summary = nodeLog.AgentVersion
-			result.Payload = nodeLog.AgentVersion
-			return result
+	if upstream["Node/Env/Dependencies"].Status == tasks.Info {
+		nodeModuleVersions, ok = upstream["Node/Env/Dependencies"].Payload.([]NodeEnv.NodeModuleVersion)
+		if !ok {
+			return tasks.Result{
+				Status:  tasks.None,
+				Summary: "Node module version not detected",
```
Comment (uturunku1) [inline tasks/node/agent/version.go:55]: Similar here: "Node Modules not detected. This task did not run"

File: tasks/node/agent/version.go [hunk 3]
```diff
@@ -36,63 +38,46 @@ func (t NodeAgentVersion) Dependencies() []string {
...
-			logElementNode.MatchFound = false
-		} else {
-			logElementNode.Logfile = logElement.FilePath + logElement.FileName
-			logElementNode.AgentVersion = logAgentVersion
-			logElementNode.MatchFound = true
+	agentVersion := getNodeVerFromPayload(nodeModuleVersions)
+	if agentVersion == "" {
+		return tasks.Result{
+			Status:  tasks.Warning,
+			Summary: "Node Agent Module not found for newrelic",
```
Comment (uturunku1) [inline tasks/node/agent/version.go:64]: Maybe we should add a more explicit message to the user, like:  "We were unable to find the 'newrelic' module required for the Node Agent installation. Make sure to run 'npm install newrelic' and verify that 'newrelic' is listed in your package.json"

**Review by uturunku1 — CHANGES_REQUESTED** [review]
Loooking goood! I requested a few changes. I also noticed that we still have the `logNodeAgentVersion` struct defined at the top of the file version.go. If we are not using that struct anywhere anymore, let's get rid of it.

File: tasks/node/agent/version.go (hunk 3 above)
Comment (KayLotura2) [reply tasks/node/agent/version.go:64]: That's a great message. Thanks for the suggestion.

File: tasks/node/agent/version.go [hunk 4]
```diff
@@ -29,70 +25,51 @@ func (t NodeAgentVersion) Explain() string {
 // Dependencies - Returns the dependencies for this task.
 func (t NodeAgentVersion) Dependencies() []string {
 	return []string{
-		"Node/Config/Agent",
-		"Base/Log/Copy",
+		"Node/Env/Dependencies",
```
Comment (uturunku1) [inline tasks/node/agent/version.go:28]: We were missing listing this relevant dependency prior to verifying its status and payload.

File: tasks/node/agent/version.go [hunk 5]
```diff
@@ -29,70 +25,51 @@ func (t NodeAgentVersion) Explain() string {
...
 
 // Execute - The core work within this task
 func (t NodeAgentVersion) Execute(options tasks.Options, upstream map[string]tasks.Result) tasks.Result {
-	var result tasks.Result
-
-	if upstream["Node/Config/Agent"].Status != tasks.Success {
-		result.Status = tasks.None
-		result.Summary = "Node agent not detected"
-		return result
+	if upstream["Node/Env/Dependencies"].Status != tasks.Info {
```
Comment (uturunku1) [inline tasks/node/agent/version.go:34]: I removed our check for Node/Config/Agent because Node/Env/Dependencies already checks that Node/Config/Agent ran before it.

File: tasks/node/agent/version.go [hunk 6]
```diff
@@ -29,70 +25,51 @@ func (t NodeAgentVersion) Explain() string {
...
-	searchString := `Node[.]js[.] Agent version:\s*([0-9.]+);`
-	filepath := logElement.FilePath + logElement.FileName
-	agentVersion, err := tasks.ReturnLastStringSubmatchInFile(searchString, filepath)
-	if err != nil {
-		return "", err
+	log.Debug("Agent version", agentVersion)
+	return tasks.Result{
+		Status:  tasks.Info,
+		Summary: fmt.Sprintf("Node Agent Version %s found", agentVersion),
+		Payload: agentVersion,
```
Comment (uturunku1) [inline tasks/node/agent/version.go:62]: We were missing returning a Payload and while I was smoke testing it became obvious because other tasks rely on this payload, and of course, they were not running or failing.

File: tasks/node/agent/version.go [hunk 7]
```diff
@@ -29,70 +25,51 @@ func (t NodeAgentVersion) Explain() string {
...
+}
 
-	if len(agentVersion) > 0 {
-		// we've got a match, return the first capture group
-		return agentVersion[1], nil
+func getNodeVerFromPayload(payload []NodeEnv.NodeModuleVersion) string {
+	for _, nodeModuleVersion := range payload {
+		if nodeModuleVersion.Module == "newrelic" {
+			return nodeModuleVersion.Version
+		}
```
Comment (uturunku1) [inline tasks/node/agent/version.go:70]: We want to return only `nodeModuleVersion.Version` instead of `nodeModuleVersion.Module+ nodeModuleVersion.Version` because the downstream tasks that will use this information/payload only expect a string made out of digits.

File: tasks/node/agent/version_test.go [hunk 8]
```diff
@@ -15,132 +16,130 @@ func TestNodeAgentVersion(t *testing.T) {
...
 		})
-
-		Context("When given a logElement with a malformed filepath", func() {
+		Context("When given a Node Agent Module is found", func() {
 			BeforeEach(func() {
-				incomingLogs = []logtask.LogElement{
-					logtask.LogElement{
-						FileName: "newrelic_agent.BADFILE.log",
-						FilePath: "../../fixtures/node/",
+				upstream = map[string]tasks.Result{
```
Comment (uturunku1) [inline tasks/node/agent/version_test.go:112]: I added one additional test for a successful case when we do get a node agent version

File: tasks/node/env/dependencies.go [hunk 9]
```diff
@@ -44,7 +44,7 @@ func (p NodeEnvDependencies) Execute(option tasks.Options, upstream map[string]t
 	if upstream["Node/Config/Agent"].Status != tasks.Success {
 		return tasks.Result{
 			Status:  tasks.None,
-			Summary: "Node agent not detected. This task did not run",
+			Summary: "Node agent config file not detected. This task did not run",
```
Comment (uturunku1) [inline tasks/node/env/dependencies.go:47]: The following small changes are completely unrelated to this PR. It is a very nitpicky change in the summary of other tasks to make it more explicit that they did not run because we did not find a config file. The previous summary seems a little less clear to me.

**Review by uturunku1 — APPROVED** [review]
